src/ephemeris.c

```diff
@@ -228,7 +213,20 @@ s8 calc_sat_state(const ephemeris_t *e, const gps_time_t *t,
   assert(clock_err != NULL);
   assert(clock_rate_err != NULL);
   assert(e != NULL);
-  assert(ephemeris_valid(e, t));
+
+  /* If dt is greater than fit_interval hours our ephemeris isn't valid. */
+  double dt = gpsdifftime(t, &e->toe);
```

Comment:
Yes but it was the only way to get a warning in here but not in ephemeris_valid about using the old ephemeris. Maybe I will just update the other warning to indicate it could be either in invalid ephemeris, or a too old ephemeris.